### PR TITLE
Add simple-hub package

### PR DESCRIPTION
**Pull request by kerzhner**
Current functionality:
1. Listen for firebase messages.
1. Add signature to incoming message.
2. Send message back to sender.

Todo:
- [x] Listen to asset holder and match deposits.
- [x] Support n participant channels.

### CODE REVIEW

File: packages/simple-hub/src/wallet/index.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import {cHubStateChannelPK, cHubStateChannelAddress} from '../constants';
+import {SignedState, Message, Participant} from '@statechannels/xstate-wallet/lib/src/store/types';
+
+function broadcastRecipents(participants: Participant[]): Participant[] {
+  return participants.filter(p => p.participantId !== cHubStateChannelAddress);
+}
+
+export function respondToMessage(wireMessage: WireMessage): WireMessage[] {
+  const message = deserializeMessage(wireMessage);
+  const lastState = R.last(message.signedStates);
```
Comment (andrewgordstewart) [inline packages/simple-hub/src/wallet/index.ts]: Why should this only sign the last state?

Should this check that the hub is a participant in the channel?

File: packages/simple-hub/.env.production [hunk 2]
```diff
@@ -0,0 +1,12 @@
+FIREBASE_PROJECT=rock-paper-scissors-production
```
Comment (snario) [inline packages/simple-hub/.env.production]: Do we need to begin this new package with Firebase baked in?

File: packages/simple-hub/.eslintrc.js [hunk 3]
```diff
@@ -0,0 +1,38 @@
+const baseConfig = require('../../.eslintrc.js');
```
Comment (snario) [inline packages/simple-hub/.eslintrc.js]: Like some other configs, you don't need to use `require` to inherit from the monorepo root config file. You can add `../../.eslintrc` to `extends` of the exported object.

File: packages/simple-hub/.vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,60 @@
+{
```
Comment (snario) [inline packages/simple-hub/.vscode/launch.json]: Consider adding some of this to the monorepo root `launch.json`?

File: packages/simple-hub/env.js [hunk 5]
```diff
@@ -0,0 +1,55 @@
+'use strict';
+
+// FIXME: To remove @statechannels/hub's dependency on @statechannels/devtools,
```
Comment (snario) [inline packages/simple-hub/env.js]: This FIXME is only here to avoid needing this dependency for `hub` on Heroku. You probably should just use `@statechannels/devtools` for now on this package? Unless you want to get this on Heroku right now to test it against the `xstate-wallet` `rps`.

File: packages/simple-hub/src/wallet/__test__/ledger-response.json [hunk 6]
```diff
@@ -0,0 +1,55 @@
+[
```
Comment (snario) [inline packages/simple-hub/src/wallet/__test__/ledger-response.json]: Can you auto-generate the test message data?

File: packages/simple-hub/src/wallet/index.ts [hunk 7]
```diff
@@ -0,0 +1,23 @@
+import {Message as WireMessage} from '@statechannels/wire-format';
+import * as R from 'ramda';
+import {signState} from '@statechannels/xstate-wallet/lib/src/store/state-utils';
+import {deserializeMessage} from '@statechannels/xstate-wallet/lib/src/serde/wire-format/deserialize';
+import {serializeMessage} from '@statechannels/xstate-wallet/lib/src/serde/wire-format/serialize';
+import {cHubStateChannelPK, cHubStateChannelAddress} from '../constants';
+import {SignedState, Message, Participant} from '@statechannels/xstate-wallet/lib/src/store/types';
+
+function broadcastRecipents(participants: Participant[]): Participant[] {
```
Comment (snario) [inline packages/simple-hub/src/wallet/index.ts]: Typo: `broadcastRecipents` -> `broadcastRecipients`

File: packages/simple-hub/src/wallet/index.ts [hunk 8]
```diff
@@ -0,0 +1,23 @@
+import {Message as WireMessage} from '@statechannels/wire-format';
+import * as R from 'ramda';
+import {signState} from '@statechannels/xstate-wallet/lib/src/store/state-utils';
```
Comment (snario) [inline packages/simple-hub/src/wallet/index.ts]: You should avoid doing file-level imports if possible.

File: packages/xstate-wallet/package.json [hunk 9]
```diff
@@ -1,5 +1,5 @@
 {
-  "name": "xstate-wallet",
+  "name": "@statechannels/xstate-wallet",
```
Comment (snario) [inline packages/xstate-wallet/package.json]: This could possibly be a separate PR? Not necessary though.

File: packages/simple-hub/readme.md [hunk 10]
```diff
@@ -0,0 +1,97 @@
+# Unit testing and running the hub locally
+
+## Setup
+
+### postgresql
```
[the remark this answers is not in the document]
Comment (kerzhner) [reply packages/simple-hub/readme.md]: ec4a0387

File: packages/simple-hub/readme.md [hunk 11]
```diff
@@ -0,0 +1,97 @@
...
+yarn test:ci
+```
+
+## Deploying
+
+Heroku runs a production version of the build `Dockerfile.hub.staging` in the root of the repo. To create a deployment you must:
+
+**Build the Dockerfile locally, by running**
+```bash
+docker build -t registry.heroku.com/statechannels-hub-staging/statechannels-hub -f Dockerfile.hub.staging .
```
[the remark this answers is not in the document]
Comment (kerzhner) [reply packages/simple-hub/readme.md]: 17c6a88d

File: packages/simple-hub/src/wallet/index.ts (hunk 1 above)
Comment (kerzhner) [reply packages/simple-hub/src/wallet/index.ts]: It's a good idea to check if the hub is a participant. Which states should the hub sign? All states?

File: packages/simple-hub/.env.production (hunk 2 above)
Comment (kerzhner) [reply packages/simple-hub/.env.production]: Would you mind clarifying? Do you mean defining the firebase project as a constant instead of an environment variable?

File: packages/simple-hub/.eslintrc.js (hunk 3 above)
Comment (kerzhner) [reply packages/simple-hub/.eslintrc.js]: Good call. 84842e51

File: packages/simple-hub/.vscode/launch.json (hunk 4 above)
Comment (kerzhner) [reply packages/simple-hub/.vscode/launch.json]: https://github.com/statechannels/monorepo/issues/1083

File: packages/simple-hub/env.js (hunk 5 above)
Comment (kerzhner) [reply packages/simple-hub/env.js]: Got it. It's likely that the `simple-hub` will be on Heroku shortly.

File: packages/simple-hub/src/wallet/__test__/ledger-response.json (hunk 6 above)
Comment (kerzhner) [reply packages/simple-hub/src/wallet/__test__/ledger-response.json]: Yep, we can. Probably worth doing down the line.

File: packages/simple-hub/src/wallet/index.ts (hunk 7 above)
Comment (kerzhner) [reply packages/simple-hub/src/wallet/index.ts]: 38047992

File: packages/xstate-wallet/package.json (hunk 9 above)
Comment (kerzhner) [reply packages/xstate-wallet/package.json]: Tempted to leave this change in since it doesn't affect other packages.

File: packages/simple-hub/src/wallet/index.ts (hunk 1 above)
Comment (kerzhner) [reply packages/simple-hub/src/wallet/index.ts]: From @andrewgordstewart in Slack: "any state in a channel with the hub"

File: packages/simple-hub/src/wallet/__test__/ledger-response.json (hunk 6 above)
Comment (kerzhner) [reply packages/simple-hub/src/wallet/__test__/ledger-response.json]: 581b0669

File: packages/simple-hub/src/wallet/index.ts (hunk 1 above)
Comment (kerzhner) [reply packages/simple-hub/src/wallet/index.ts]: c2769939fd2c9da8925c9d9a4210c49cf21955b9

File: packages/simple-hub/.env.production (hunk 2 above)
Comment (kerzhner) [reply packages/simple-hub/.env.production]: Or do you mean abstracting away firebase into a seperate layer and/or package?
